ledger/complete/mtrie/flattener/iterator.go

```diff
@@ -53,6 +53,15 @@ type NodeIterator struct {
 	// This has the advantage, that we gracefully handle tries whose root node is nil.
 	unprocessedRoot *node.Node
 	stack           []*node.Node
+	// visitedNodes are nodes that were visited and can be skipped during
+	// traversal through dig(). visitedNodes is used to optimize node traveral
+	// IN FOREST by skipping nodes in shared sub-tries after they are visited,
+	// because sub-tries are shared between tries (original MTrie before register updates
+	// and updated MTrie after register writes).
+	// NodeIterator only uses visitedNodes for read operation.
+	// No special handling is needed if visitedNodes is nil.
+	// WARNING: visitedNodes is not safe for concurrent use.
```

Comment:
could you add this line also to the existing `NewNodeIterator` constructor please
https://github.com/onflow/flow-go/blob/1ebde088e9e978de4fc3a522168f349d9eb6eb15/ledger/complete/mtrie/flattener/iterator.go#L77

Reply:
I'm not sure we should add the same concurrency warning here because `visitedNodes` will always be `nil` which appears to make it safe for concurrent use (unless I'm mistaken).

I [added this comment](https://github.com/onflow/flow-go/pull/1944/commits/d3a32c6eae1a1606f90a837876f7699765c3e550) to `NewNodeIterator`:

```Go
// NodeIterator created by NewNodeIterator is safe for concurrent use
// because visitedNodes is always nil in this case.
```

Please let me know if I need to update the comment.  🙏